src/app-shared/core/interface/common.interface.ts

```diff
@@ -12,3 +14,10 @@ export enum HttpStatusCode {
   NOT_FOUND = 404,
   INTERNAL_SERVER_ERROR = 500,
 }
+
+export interface ICRUDService<T> {
+  get(id: string): Observable<T>;
+  post(object: T): Observable<T>;
+  put(object: T): Observable<T>;
+  delete(id: string): Observable<T>;
```

Comment:
This interface should have `create/read/update/delete` (as in `C.R.U.D`). not `get/post/put` and the `delete` method should return void.